Util_UsageStatistics.php

```diff
@@ -1,218 +1,402 @@
+		}
 	}
 
-
-
-	static public function bytes_to_size2( $a, $p1, $p2 = null, $p3 = null ) {
+	/**
+	 * Converts a nested byte value from an array into a human-readable size format (KB, MB, or GB).
+	 *
+	 * This method retrieves a nested value from an array using the `v` method, then converts it
+	 * into a human-readable size format using the `bytes_to_size` method. If the value is null,
+	 * it returns 'n/a'.
+	 *
+	 * @param array           $a  The input array to search for the byte value.
+	 * @param string|int      $p1 The first key to search for.
+	 * @param string|int|null $p2 The second key to search for (optional).
+	 * @param string|int|null $p3 The third key to search for (optional).
+	 *
+	 * @return string The formatted size (e.g., '1.2 GB', '512.0 MB', or '256.0 KB'),
+	 *                or 'n/a' if the value is null or not found.
+	 */
+	public static function bytes_to_size2( $a, $p1, $p2 = null, $p3 = null ) {
 		$v = self::v( $a, $p1, $p2, $p3 );
-		if ( is_null( $v ) )
+		if ( is_null( $v ) ) {
 			return 'n/a';
+		}
 
 		return self::bytes_to_size( $v );
 	}
 
-
-
-	static public function percent( $v1, $v2 ) {
-		if ( $v2 == 0 ) {
+	/**
+	 * Calculates the percentage of two values and formats it as a string with a '%' symbol.
+	 *
+	 * If the denominator is zero, the result is '0 %'. If the numerator exceeds the denominator,
+	 * the result is capped at '100 %'. Otherwise, the percentage is calculated and formatted
+	 * as an integer.
+	 *
+	 * @param float|int $v1 The numerator value.
+	 * @param float|int $v2 The denominator value.
+	 *
+	 * @return string The calculated percentage as a string (e.g., '75 %'), or '0 %' if $v2 is 0.
+	 */
+	public static function percent( $v1, $v2 ) {
+		if ( 0 === $v2 ) {
 			return '0 %';
-		} elseif ($v1 > $v2 ) {
+		} elseif ( $v1 > $v2 ) {
 			return '100 %';
 		} else {
 			return sprintf( '%d', $v1 / $v2 * 100 ) . ' %';
 		}
 	}
 
-
-
-	static public function percent2( $a, $property1, $property2 ) {
-		if ( !isset( $a[$property1] ) || !isset( $a[$property2] ) )
+	/**
+	 * Calculates the percentage of two properties from an array and formats it as a string with a '%' symbol.
+	 *
+	 * If either property is missing or the denominator is zero, a fallback value is returned.
+	 *
+	 * @param array      $a         The input array containing the properties.
+	 * @param string|int $property1 The key for the numerator value.
+	 * @param string|int $property2 The key for the denominator value.
+	 *
+	 * @return string The calculated percentage as a string (e.g., '75 %'), 'n/a' if properties
+	 *                are missing, or '0 %' if $a[$property2] is 0.
+	 */
+	public static function percent2( $a, $property1, $property2 ) {
+		if ( ! isset( $a[ $property1 ] ) || ! isset( $a[ $property2 ] ) ) {
 			return 'n/a';
-		else if ( $a[$property2] == 0 )
-				return '0 %';
-			else
-				return sprintf( '%d', $a[$property1] / $a[$property2] * 100 ) . ' %';
+		} elseif ( 0 === $a[ $property2 ] ) {
```

Comment:
The value may not be an `int` -- change to `} elseif ( empty( $a[ $property2 ] ) ) {`.

Reply:
Updated